DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -161,34 +165,41 @@ function Test-TargetResource
             -InterfaceAlias $InterfaceAlias
     }
 
-    #Get the current DNS Server Addresses based on the parameters given.
-    $currentAddress = (Get-DnsClientServerAddress `
-        -InterfaceAlias $InterfaceAlias `
-        -AddressFamily $AddressFamily `
-        -ErrorAction Stop).ServerAddresses
-
-    #Check if the Server addresses are the same as the desired addresses.
-    [Boolean] $addressDifferent = (@(Compare-Object `
-            -ReferenceObject $currentAddress `
-            -DifferenceObject $Address `
-            -SyncWindow 0).Length -gt 0)
-
-    if ($addressDifferent)
+    # Get the current DNS Server Addresses based on the parameters given.
+    # Interface Alias may return multiple when passed WildCard (*)
+    $interfaceAliases = Get-NetAdapter -Name $InterfaceAlias
+    foreach ($alias in $interfaceAliases.Name)
     {
-        $desiredConfigurationMatch = $false
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.DNSServersNotCorrectMessage) `
-                -f ($Address -join ','),($currentAddress -join ',')
-            ) -join '' )
-    }
-    else 
-    { 
-        #Test will return true in this case
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.DNSServersSetCorrectlyMessage)
-            ) -join '' )
+        $currentAddress = (Get-DnsClientServerAddress `
+            -InterfaceAlias $alias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop).ServerAddresses
+
+        # Check if the Server addresses are the same as the desired addresses.
+        [Boolean] $addressDifferent = (@(Compare-Object `
+                -ReferenceObject $currentAddress `
+                -DifferenceObject $Address `
+                -SyncWindow 0).Length -gt 0)
+
+        if ($addressDifferent)
+        {
+            $desiredConfigurationMatch.Add($false)
+            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+                $($LocalizedData.DNSServersNotCorrectMessage) `
+                    -f ($Address -join ','),($currentAddress -join ',')
+                ) -join '' )
```

Comment:
One thought woudl be to just return $false here and exit out of the loop, as we're already going to need to make changes not sure it's really worth the time to check everything.